File: src/dotnet/Core/Domain/Entities/ParsedQuery.cs
using Rag.Candidates.Core.Domain.Enums;

namespace Rag.Candidates.Core.Domain.Entities;

public sealed class ParsedQuery
{
    public required string QueryText { get; init; }
    public required QueryIntent QueryIntent { get; init; }
    public required string[] RequiredTechnologies { get; init; }
    public SeniorityLevel? MinSeniorityLevel { get; init; }
    public int? MinYearsExperience { get; init; }
}

File: src/python/core/domain/entities/parsed_query.py
from dataclasses import dataclass
from typing import List, Optional
from ..enums.seniority_level import SeniorityLevel
from ..enums.query_intent import QueryIntent


@dataclass(frozen=True)
class ParsedQuery:
    query_text: str
    query_intent: QueryIntent
    required_technologies: List[str]
    min_seniority_level: Optional[SeniorityLevel]
    min_years_experience: Optional[int]
